### PR TITLE
Simplify slider_blockers method

### PR DESCRIPTION
**Pull request by pb00068**
Now that classical evaluation was removed, we can adopt this method to the needs of set_check_info.

no functional change

### CODE REVIEW

**Comment by pb00068** [comment]
N.B.: a templated version of simplified slider_blockers is being tested here at https://tests.stockfishchess.org/tests/view/64de132587ff9c9f7a9351f6 but I doubts its a good idea to make this templated by color.

**Comment by vondele** [comment]
might be good to test this patch as is on fishtest, to be sure? (just STC is fine)

**Comment by Sopel97** [comment]
I suggest renaming it to `update_slider_blockers`

File: src/position.cpp [hunk 1]
```diff
@@ -436,37 +436,34 @@ string Position::fen() const {
   return ss.str();
 }
 
+/// Position::slider_blockers() calculates
```
Comment (Sopel97) [inline src/position.cpp:439]: name divergence

**Comment by pb00068** [comment]
> might be good to test this patch as is on fishtest, to be sure? (just STC is fine)

https://tests.stockfishchess.org/tests/view/64e70b899009777747557b43
:2.95 (-2.94,2.94) <-1.75,0.25>
Total: 298176 W: 75802 L: 75868 D: 146506
Ptnml(0-2): 908, 33608, 80192, 33402, 978

**Comment by snicolet** [comment]
merged via 1f7ff8406d323e634a2aa1e1264042340707cdd9, thanks :-)

**Pull request closed without merging**